src/main/java/com/ubb/zenith/service/UserPlaylistService.java

```diff
@@ -0,0 +1,122 @@
+package com.ubb.zenith.service;
+
+import com.ubb.zenith.exception.PlaylistNotFoundException;
+import com.ubb.zenith.exception.UserNotFoundException;
+import com.ubb.zenith.model.User;
+import com.ubb.zenith.model.Playlist;
+import com.ubb.zenith.model.UserPlaylist;
+import com.ubb.zenith.repository.UserPlaylistRepository;
+import com.ubb.zenith.repository.UserRepository;
+import com.ubb.zenith.repository.PlaylistRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+import java.util.List;
+
+@Service
+public class UserPlaylistService {
+
+    @Autowired
+    private UserPlaylistRepository userPlaylistRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PlaylistRepository playlistRepository;
+
+    /**
+     * Retrieves all user playlists from the repository.
+     *
+     * @return a list of all user playlists.
+     */
+    public List<UserPlaylist> getAll() {
+        return userPlaylistRepository.findAll();
+    }
+
+    /**
+     * Adds a new user playlist.
+     *
+     * @param userId the ID of the user.
+     * @param playlistId the ID of the playlist.
+     * @param date the date when the playlist was added to the user.
+     * @return the added user playlist.
+     * @throws UserNotFoundException if the user is not found.
+     * @throws PlaylistNotFoundException if the playlist is not found.
+     */
+    public UserPlaylist add(Integer userId, Integer playlistId, Date date) throws UserNotFoundException, PlaylistNotFoundException {
```

Comment:
We can make UserPlaylistDTO with these fields: username, name (the name of the playlist) and date
We can search the users and playlists by name, not by id